### ISSUE TITLE
build failed

### ISSUE THREAD

**Issue by molele2** [cocaine/cocaine-core#206]
Scanning dependencies of target json
[  2%] Building CXX object CMakeFiles/json.dir/foreign/jsoncpp-0.6.0-rc2/src/lib_json/json_value.cpp.o
[  5%] Building CXX object CMakeFiles/json.dir/foreign/jsoncpp-0.6.0-rc2/src/lib_json/json_reader.cpp.o
[  8%] Building CXX object CMakeFiles/json.dir/foreign/jsoncpp-0.6.0-rc2/src/lib_json/json_writer.cpp.o
Linking CXX static library libjson.a
[  8%] Built target json
Scanning dependencies of target cocaine-core
[ 11%] Building CXX object CMakeFiles/cocaine-core.dir/src/actor.cpp.o
In file included from /usr/local/include/msgpack/type.hpp:17,
                 from /usr/local/include/msgpack/object.hpp:808,
                 from /usr/local/include/msgpack.hpp:18,
                 from /storm/cocaine-core-0.11/include/cocaine/traits.hpp:24,
                 from /storm/cocaine-core-0.11/include/cocaine/rpc/slot.hpp:26,
                 from /storm/cocaine-core-0.11/include/cocaine/rpc/slots/function.hpp:27,
                 from /storm/cocaine-core-0.11/include/cocaine/rpc/slots/blocking.hpp:24,
                 from /storm/cocaine-core-0.11/include/cocaine/dispatch.hpp:26,
                 from /storm/cocaine-core-0.11/src/actor.cpp:32:
/usr/local/include/msgpack/adaptor/vector.hpp: In member function ‘void msgpack::v1::adaptor::object_with_zone<std::vector<T, std::allocator<_Tp1> >, void>::operator()(msgpack::v1::object::with_zone&, const std::vector<T, std::allocator<_Tp1> >&) const’:
/usr/local/include/msgpack/adaptor/vector.hpp:84: error: #pragma GCC diagnostic not allowed inside functions
/usr/local/include/msgpack/adaptor/vector.hpp:85: error: #pragma GCC diagnostic not allowed inside functions
/usr/local/include/msgpack/adaptor/vector.hpp:89: error: #pragma GCC diagnostic not allowed inside functions
In file included from /storm/cocaine-core-0.11/include/cocaine/rpc/channel.hpp:27,
                 from /storm/cocaine-core-0.11/src/actor.cpp:37:
/storm/cocaine-core-0.11/include/cocaine/rpc/decoder.hpp: In member function ‘size_t cocaine::io::decoder< <template-parameter-1-1> >::on_event(const char_, size_t)’:
/storm/cocaine-core-0.11/include/cocaine/rpc/decoder.hpp:91: error: no matching function for call to ‘unpack(const char_&, size_t&, size_t_&, msgpack::v1::zone_&, msgpack::v1::object_&)’
/usr/local/include/msgpack/unpack.hpp:1530: note: candidates are: msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, size_t&, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1554: note:                 msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, size_t&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1562: note:                 msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1570: note:                 msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1579: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, size_t&, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1606: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, size_t&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1614: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1622: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1632: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, size_t&, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1656: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, size_t&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1665: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1674: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1686: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked_, const char_, size_t, size_t_, bool_, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/storm/cocaine-core-0.11/include/cocaine/rpc/decoder.hpp: In member function ‘size_t cocaine::io::decoder< <template-parameter-1-1> >::on_event(const char_, size_t) [with Stream = cocaine::io::readable_streamcocaine::io::socket<cocaine::io::tcp >]’:
/storm/cocaine-core-0.11/include/cocaine/rpc/decoder.hpp:60:   instantiated from ‘void cocaine::io::decoder< <template-parameter-1-1> >::bind(MessageHandler, ErrorHandler) [with MessageHandler = std::_Bind<std::_Mem_fn<void (cocaine::actor_t::_)(int, const cocaine::io::message_t&)>(cocaine::actor_t_, int, std::_Placeholder<1>)>, ErrorHandler = std::_Bind<std::_Mem_fn<void (cocaine::actor_t::_)(int, const std::error_code&)>(cocaine::actor_t_, int, std::_Placeholder<1>)>, Stream = cocaine::io::readable_streamcocaine::io::socket<cocaine::io::tcp >]’
/storm/cocaine-core-0.11/src/actor.cpp:261:   instantiated from here
/storm/cocaine-core-0.11/include/cocaine/rpc/decoder.hpp:91: error: no matching function for call to ‘unpack(const char_&, size_t&, size_t_, msgpack::v1::zone_, msgpack::v1::object_)’
/usr/local/include/msgpack/unpack.hpp:1530: note: candidates are: msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, size_t&, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1554: note:                 msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, size_t&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1562: note:                 msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1570: note:                 msgpack::v1::unpacked msgpack::v1::unpack(const char_, size_t, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1579: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, size_t&, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1606: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, size_t&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1614: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1622: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked&, const char_, size_t, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1632: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, size_t&, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1656: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, size_t&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1665: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, bool&, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1674: note:                 msgpack::v1::object msgpack::v1::unpack(msgpack::v1::zone&, const char_, size_t, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
/usr/local/include/msgpack/unpack.hpp:1686: note:                 void msgpack::v1::unpack(msgpack::v1::unpacked_, const char_, size_t, size_t_, bool_, bool (_)(msgpack::v1::type::object_type, size_t, void_), void_, const msgpack::v1::unpack_limit&)
In file included from /storm/cocaine-core-0.11/src/actor.cpp:39:
/storm/cocaine-core-0.11/include/cocaine/traits/literal.hpp: In static member function ‘static void cocaine::io::type_traits<cocaine::io::literal, void>::pack(msgpack::v1::packer<Stream>&, const cocaine::io::literal&) [with Stream = msgpack::v1::sbuffer]’:
/storm/cocaine-core-0.11/include/cocaine/traits/typelist.hpp:127:   instantiated from ‘static void cocaine::io::type_traits<T, typename std::enable_if<boost::mpl::is_sequence::value, void>::type>::pack_sequence(msgpack::v1::packer<Stream>&, const Head&, const Tail& ...) [with It = boost::mpl::l_iterboost::mpl::l_item<mpl_::long_<1l, std::basic_string<char, std::char_traits<char>, std::allocator<char> >, boost::mpl::l_end> >, Stream = msgpack::v1::sbuffer, Head = cocaine::io::literal, Tail = , T = boost::mpl::list<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>]’
/storm/cocaine-core-0.11/include/cocaine/traits/typelist.hpp:78:   instantiated from ‘static void cocaine::io::type_traits<T, typename std::enable_if<boost::mpl::is_sequence::value, void>::type>::pack(msgpack::v1::packer<Stream>&, const Args& ...) [with Stream = msgpack::v1::sbuffer, Args = cocaine::io::literal, T = boost::mpl::list<std::basic_string<char, std::char_traits<char>, std::allocator<char> >, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na, mpl_::na>]’
/storm/cocaine-core-0.11/include/cocaine/rpc/encoder.hpp:80:   instantiated from ‘void cocaine::io::encoder< <template-parameter-1-1> >::write(uint64_t, Args&& ...) [with Event = cocaine::io::rpc::chunk, Args = cocaine::io::literal, Stream = cocaine::io::writable_streamcocaine::io::socket<cocaine::io::tcp >]’
/storm/cocaine-core-0.11/src/actor.cpp:86:   instantiated from here
/storm/cocaine-core-0.11/include/cocaine/traits/literal.hpp:60: error: ‘class msgpack::v1::packermsgpack::v1::sbuffer’ has no member named ‘pack_raw’
/storm/cocaine-core-0.11/include/cocaine/traits/literal.hpp:61: error: ‘class msgpack::v1::packermsgpack::v1::sbuffer’ has no member named ‘pack_raw_body’
cc1plus: warnings being treated as errors
/boost/boost_1_58_0/boost/system/error_code.hpp: At global scope:
/boost/boost_1_58_0/boost/system/error_code.hpp:221: error: ‘boost::system::posix_category’ defined but not used
/boost/boost_1_58_0/boost/system/error_code.hpp:222: error: ‘boost::system::errno_ecat’ defined but not used
/boost/boost_1_58_0/boost/system/error_code.hpp:223: error: ‘boost::system::native_ecat’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:258: error: ‘boost::asio::error::system_category’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:260: error: ‘boost::asio::error::netdb_category’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:262: error: ‘boost::asio::error::addrinfo_category’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:264: error: ‘boost::asio::error::misc_category’ defined but not used
make[2]: **\* [CMakeFiles/cocaine-core.dir/src/actor.cpp.o] Error 1
make[1]: **\* [CMakeFiles/cocaine-core.dir/all] Error 2
make: **\* [all] Error 2

**Comment by 3Hren** [comment]
Platform, versions, branch?

**Comment by molele2** [comment]
for centos6.5 gcc 4.4
I try to build some library cocaine-core ,like this

**Comment by 3Hren** [comment]
Seems like it's `msgpack` incompatibility. We depend on 0.* versions, 1.* are not supported yet.

**Comment by molele2** [comment]
o,I need build mspack ? version

**Comment by 3Hren** [comment]
For example:

```bash
esafronov@cocs01h:~$ dpkg -l | grep msgpack
ii  libmsgpack3                            0.5.4-2                           binary-based efficient object serialization library
ii  msgpack-python
```

**Comment by molele2** [comment]
Which version for boost?

**Comment by 3Hren** [comment]
I think this doesn't matter, they rarely break backward compatibility. We've tested on 1.46, 1.53, 1.58.

**Comment by molele2** [comment]
Are you sure msgpack,Use 0.5.4 version
pp  erlang  example  haskell  java  msgpack  perl  php  python  README.md  ruby  test
[root@iZ947yjsmw8Z msgpack-c-cpp-0.5.4]# ls cpp
AUTHORS    ChangeLog     COPYING   LICENSE      msgpack_vc8.sln     msgpack_vc.postbuild.bat  preprocess  src
bootstrap  configure.in  Doxyfile  Makefile.am  msgpack_vc8.vcproj  NOTICE                    README.md   test
[root@iZ947yjsmw8Z msgpack-c-cpp-0.5.4]# cd cpp
[root@iZ947yjsmw8Z cpp]# ./bootstrap
./preprocess: line 4: ruby: command not found

** preprocess failed **

[root@iZ947yjsmw8Z cpp]#

**Comment by molele2** [comment]
very horrible

**Comment by molele2** [comment]
How to set cmake paramter for compile?? boost use gun99? stdc11???????????

canning dependencies of target json
[  2%] Building CXX object CMakeFiles/json.dir/foreign/jsoncpp-0.6.0-rc2/src/lib_json/json_value.cpp.o
[  5%] Building CXX object CMakeFiles/json.dir/foreign/jsoncpp-0.6.0-rc2/src/lib_json/json_reader.cpp.o
[  8%] Building CXX object CMakeFiles/json.dir/foreign/jsoncpp-0.6.0-rc2/src/lib_json/json_writer.cpp.o
Linking CXX static library libjson.a
[  8%] Built target json
Scanning dependencies of target cocaine-core
[ 11%] Building CXX object CMakeFiles/cocaine-core.dir/src/actor.cpp.o
[ 14%] Building CXX object CMakeFiles/cocaine-core.dir/src/api.cpp.o
[ 17%] Building CXX object CMakeFiles/cocaine-core.dir/src/app.cpp.o
[ 20%] Building CXX object CMakeFiles/cocaine-core.dir/src/archive.cpp.o
[ 22%] Building CXX object CMakeFiles/cocaine-core.dir/src/context.cpp.o
/storm/cocaine-core-0.11/src/context.cpp: In constructor ‘cocaine::config_t::config_t(const std::string&)’:
/storm/cocaine-core-0.11/src/context.cpp:161: error: ambiguous overload for ‘operator=’ in ‘((cocaine::config_t*)this)->cocaine::config_t::network.cocaine::config_t::<anonymous struct>::gateway = {Json::Value::asString() const(), ((Json::Value*)((Json::Value*)root.Json::Value::operator[](((const char*)"network")))->Json::Value::operator[](((const char*)"gateway")))->Json::Value::operator[](((const char*)"args"))}’
/boost/boost_1_58_0/boost/optional/optional.hpp:940: note: candidates are: boost::optional<T>& boost::optional<T>::operator=(typename boost::optional_detail::optional_base<T>::argument_type) [with T = cocaine::config_t::component_t]
/boost/boost_1_58_0/boost/optional/optional.hpp:948: note:                 boost::optional<T>& boost::optional<T>::operator=(typename boost::optional_detail::optional_base<T>::rval_reference_type) [with T = cocaine::config_t::component_t]
/boost/boost_1_58_0/boost/optional/optional.hpp:959: note:                 boost::optional<T>& boost::optional<T>::operator=(boost::none_t) [with T = cocaine::config_t::component_t]
In file included from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/bits/unique_ptr.h:41,
                 from /usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/memory:67,
                 from /storm/cocaine-core-0.11/include/cocaine/common.hpp:32,
                 from /storm/cocaine-core-0.11/include/cocaine/context.hpp:24,
                 from /storm/cocaine-core-0.11/src/context.cpp:21:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple: In constructor ‘std::_Head_base<_Idx, _Head, false>::_Head_base(_UHead&&) [with _UHead = std::_Head_base<0ul, unsigned int, false>, long unsigned int _Idx = 0ul, _Head = short unsigned int]’:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:179:   instantiated from ‘std::_Tuple_impl<_Idx, _Head, _Tail ...>::_Tuple_impl(std::_Tuple_impl<_Idx, _UElements ...>&&) [with _UElements = unsigned int, unsigned int, long unsigned int _Idx = 0ul, _Head = short unsigned int, _Tail = short unsigned int]’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:338:   instantiated from ‘std::tuple<_T1, _T2>::tuple(std::tuple<_U1, _U2>&&) [with _U1 = unsigned int, _U2 = unsigned int, _T1 = short unsigned int, _T2 = short unsigned int]’
/boost/boost_1_58_0/boost/optional/optional.hpp:619:   instantiated from ‘void boost::optional_detail::optional_base<T>::assign_expr_to_initialized(Expr&&, const void*) [with Expr = std::tuple<unsigned int, unsigned int>, T = std::tuple<short unsigned int, short unsigned int>]’
/boost/boost_1_58_0/boost/optional/optional.hpp:442:   instantiated from ‘void boost::optional_detail::optional_base<T>::assign_expr(Expr&&, const ExprPtr*) [with Expr = std::tuple<unsigned int, unsigned int>, ExprPtr = std::tuple<unsigned int, unsigned int>, T = std::tuple<short unsigned int, short unsigned int>]’
/boost/boost_1_58_0/boost/optional/optional.hpp:883:   instantiated from ‘typename boost::disable_if_c<(boost::is_base_of::value || boost::is_same::value), boost::optional<T>&>::type boost::optional<T>::operator=(Expr&&) [with Expr = std::tuple<unsigned int, unsigned int>, T = std::tuple<short unsigned int, short unsigned int>]’
/storm/cocaine-core-0.11/src/context.cpp:147:   instantiated from here
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:94: error: cannot convert ‘std::_Head_base<0ul, unsigned int, false>’ to ‘short unsigned int’ in initialization
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple: In constructor ‘std::_Head_base<_Idx, _Head, false>::_Head_base(_UHead&&) [with _UHead = std::_Head_base<1ul, unsigned int, false>, long unsigned int _Idx = 1ul, _Head = short unsigned int]’:
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:179:   instantiated from ‘std::_Tuple_impl<_Idx, _Head, _Tail ...>::_Tuple_impl(std::_Tuple_impl<_Idx, _UElements ...>&&) [with _UElements = unsigned int, long unsigned int _Idx = 1ul, _Head = short unsigned int, _Tail = ]’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:179:   instantiated from ‘std::_Tuple_impl<_Idx, _Head, _Tail ...>::_Tuple_impl(std::_Tuple_impl<_Idx, _UElements ...>&&) [with _UElements = unsigned int, unsigned int, long unsigned int _Idx = 0ul, _Head = short unsigned int, _Tail = short unsigned int]’
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:338:   instantiated from ‘std::tuple<_T1, _T2>::tuple(std::tuple<_U1, _U2>&&) [with _U1 = unsigned int, _U2 = unsigned int, _T1 = short unsigned int, _T2 = short unsigned int]’
/boost/boost_1_58_0/boost/optional/optional.hpp:619:   instantiated from ‘void boost::optional_detail::optional_base<T>::assign_expr_to_initialized(Expr&&, const void*) [with Expr = std::tuple<unsigned int, unsigned int>, T = std::tuple<short unsigned int, short unsigned int>]’
/boost/boost_1_58_0/boost/optional/optional.hpp:442:   instantiated from ‘void boost::optional_detail::optional_base<T>::assign_expr(Expr&&, const ExprPtr*) [with Expr = std::tuple<unsigned int, unsigned int>, ExprPtr = std::tuple<unsigned int, unsigned int>, T = std::tuple<short unsigned int, short unsigned int>]’
/boost/boost_1_58_0/boost/optional/optional.hpp:883:   instantiated from ‘typename boost::disable_if_c<(boost::is_base_of::value || boost::is_same::value), boost::optional<T>&>::type boost::optional<T>::operator=(Expr&&) [with Expr = std::tuple<unsigned int, unsigned int>, T = std::tuple<short unsigned int, short unsigned int>]’
/storm/cocaine-core-0.11/src/context.cpp:147:   instantiated from here
/usr/lib/gcc/x86_64-redhat-linux/4.4.7/../../../../include/c++/4.4.7/tuple:94: error: cannot convert ‘std::_Head_base<1ul, unsigned int, false>’ to ‘short unsigned int’ in initialization
cc1plus: warnings being treated as errors
/boost/boost_1_58_0/boost/system/error_code.hpp: At global scope:
/boost/boost_1_58_0/boost/system/error_code.hpp:221: error: ‘boost::system::posix_category’ defined but not used
/boost/boost_1_58_0/boost/system/error_code.hpp:222: error: ‘boost::system::errno_ecat’ defined but not used
/boost/boost_1_58_0/boost/system/error_code.hpp:223: error: ‘boost::system::native_ecat’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:258: error: ‘boost::asio::error::system_category’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:260: error: ‘boost::asio::error::netdb_category’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:262: error: ‘boost::asio::error::addrinfo_category’ defined but not used
/boost/boost_1_58_0/boost/asio/error.hpp:264: error: ‘boost::asio::error::misc_category’ defined but not used
make[2]: *** [CMakeFiles/cocaine-core.dir/src/context.cpp.o] Error 1
make[1]: *** [CMakeFiles/cocaine-core.dir/all] Error 2
make: *** [all] Error 2
